src/System.Diagnostics.Debug/src/System/Diagnostics/DebuggerTypeProxyAttribute.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Contracts;
+
+namespace System.Diagnostics
+{
+    [AttributeUsage(AttributeTargets.Struct | AttributeTargets.Class | AttributeTargets.Assembly, AllowMultiple = true)]
+    public sealed class DebuggerTypeProxyAttribute : Attribute
+    {
+        private string typeName;
+        private string targetName;
+        private Type target;
+
+        public DebuggerTypeProxyAttribute(Type type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException("type");
+            }
+            Contract.EndContractBlock();
+
+            this.typeName = type.AssemblyQualifiedName;
+        }
+
+        public DebuggerTypeProxyAttribute(string typeName)
+        {
+            this.typeName = typeName;
+        }
+        public string ProxyTypeName
+        {
+            get { return typeName; }
+        }
+
+        public Type Target
+        {
+            set
```

Comment:
I was confused here for a moment... makes me realize how trained I am to seeing the getter come first :)